File: BW8AGU/VB/Ch08/BW8AGU.Samples.VB/BW8AGU.Samples.VB/Common/RSSHelper.vb
Imports Windows.Web.Syndication

Public NotInheritable Class RssHelper
    Private Sub New()
    End Sub
    Public Shared Async Function GetFeedTitleAsync() As Task(Of String)
        Dim response As String = [String].Empty
        Dim feed As New SyndicationFeed()
        Dim client As New SyndicationClient()

        Try
            feed = Await client.RetrieveFeedAsync(New Uri("http://www.apress.com/index.php/dailydeals/index/rss"))

            response = feed.GetXmlDocument(SyndicationFormat.Rss20).GetXml()
        Catch ex As Exception
            Dim status As SyndicationErrorStatus = SyndicationError.GetStatus(ex.HResult)
            If status = SyndicationErrorStatus.InvalidXml Then
                response += "Invalid XML!"
            End If

            If status = SyndicationErrorStatus.Unknown Then
                response = ex.Message
            End If
        End Try

        Return response
    End Function
End Class

File: BW8AGU/C#/Ch08/BW8AGU.Samples.CSharp/Common/RssHelper.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;
using Windows.Web.Syndication;

namespace BW8AGU.Samples.CSharp.Common
{
    public static class RssHelper
    {
        public async static Task<string> GetFeedTitleAsync()
        {
            string response = String.Empty;
            SyndicationFeed feed = new SyndicationFeed();
            SyndicationClient client = new SyndicationClient();

            try
            {
                feed = await client.RetrieveFeedAsync(
                    new Uri("http://www.apress.com/index.php/dailydeals/index/rss"));

                response = feed.GetXmlDocument(SyndicationFormat.Rss20).GetXml();
            }
            catch (Exception ex)
            {
                SyndicationErrorStatus status = SyndicationError.GetStatus(ex.HResult);
                if (status == SyndicationErrorStatus.InvalidXml)
                {
                    response += "Invalid XML!";
                }

                if (status == SyndicationErrorStatus.Unknown)
                {
                    response = ex.Message;
                }
            }

            return response;
        }
    }
}
